apps/web/src/components/Publication/OpenAction/UnknownModule/Decent/CurrencySelector.tsx

```diff
@@ -0,0 +1,199 @@
+import type { FC } from 'react';
+import type { Address } from 'viem';
+
+import { STATIC_IMAGES_URL } from '@hey/data/constants';
+import getRedstonePrice from '@hey/helpers/getRedstonePrice';
+import getTokenImage from '@hey/helpers/getTokenImage';
+import { useQuery } from '@tanstack/react-query';
+import { useAllowedTokensStore } from 'src/store/persisted/useAllowedTokensStore';
+import { formatUnits } from 'viem';
+import { useAccount, useBalance } from 'wagmi';
+
+const OA_SUPPORTED_CURRENCIES = ['USDC', 'WETH', 'WMATIC'];
+
+interface CurrencySelectorProps {
+  onSelectCurrency: (currency: Address) => void;
+}
+
+const CurrencySelector: FC<CurrencySelectorProps> = ({ onSelectCurrency }) => {
+  const { address } = useAccount();
+
+  const { allowedTokens } = useAllowedTokensStore();
+
+  const { data: wmaticBalanceData, isLoading: wmaticBalanceLoading } =
+    useBalance({
+      address,
+      chainId: 137,
+      query: { refetchInterval: 10000 },
+      token: allowedTokens.find((token) => token.symbol === 'WMATIC')
+        ?.contractAddress as Address
+    });
+
+  const { data: wethBalanceData, isLoading: wethBalanceLoading } = useBalance({
+    address,
+    chainId: 137,
+    query: { refetchInterval: 10000 },
+    token: allowedTokens.find((token) => token.symbol === 'WETH')
```

Comment:
we can create a method called getToken(symbol)? also i'd say compare with token address instead of symbol which is more unique, symbol might be same on future cases

Reply:
```tsx
const OA_SUPPORTED_CURRENCY_ADDRESSES: Record<string, `0x${string}`> = {
  USDC: '0x3c499c542cEF5E3811e1192ce70d8cC03d5c3359',
  WETH: '0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619',
  WMATIC: '0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270'
};
```

using this to limit the case to only a handful of supported tokens